### PR TITLE
fix(useListNavigation): skip hidden items

### PR DESCRIPTION
**Pull request by mihkeleidast**
Skips hidden elements from list navigation. Not sure if checking `offsetParent` is the best solution, feel free to suggest some alternative.

## Background

I am building this combobox with a tree popup. When a group is collapsed, the items inside are still rendered in react (because we need to get the label/id data back to the parent component), but hidden via CSS. When they are hidden, they should be skipped by list navigation.

image

### CODE REVIEW

File: packages/react/test/setupTests.ts [hunk 1]
```diff
@@ -10,3 +10,35 @@ jest
...
+}
+
+// From https://github.com/jsdom/jsdom/issues/1261#issuecomment-512217225
+Object.defineProperty(HTMLElement.prototype, 'offsetParent', {
+    get() {
+        // eslint-disable-next-line
+        let element = this;
+        while (!isNullOrUndefined(element) &&
+                    (isNullOrUndefined(element.style) ||
+                    isNullOrUndefined(element.style.display) ||
```
Comment (atomiks) [inline packages/react/test/setupTests.ts:25]: Prettier on this?

**Comment by atomiks** [comment]
One edge case is that if the list item is `position: fixed` then it'll be marked as hidden even if it's not actually... I expect that to be extremely rare, but theoretically possible. Maybe should check if `offsetWidth`/height is `0` instead? Might be an edge case there...

**Comment by mihkeleidast** [comment]
> One edge case is that if the list item is `position: fixed` then it'll be marked as hidden even if it's not actually... I expect that to be extremely rare, but theoretically possible. Maybe should check if `offsetWidth`/height is `0` instead? Might be an edge case there...

Hmm, could we punt this change for when such usecase presents itself? Checking `offsetWidth/offsetHeight` would, for example, bring in edge cases with `visibility: hidden;` elements (as those are hidden but still take up space). Not sure we need to choose one or the other edge case to fix right now, both should be rare, especially in the list navigation context.

**Comment by atomiks** [comment]
Yeah agree it's fine enough for now
